src/utils/runtime/Cargo.toml

```diff
@@ -46,5 +46,8 @@ tracing-subscriber = { version = "0.3", features = ["fmt", "parking_lot", "std",
 opentelemetry = { version = "0.20", default-features = false, features = ["rt-tokio"] }
 workspace-hack = { path = "../../workspace-hack" }
 
+[lib]
+test = false
```

Comment:
So if future developers add unit tests for the `runtime` crate, they will silently not be run? I'm not sure if this could be a potential risk. 😕

Reply:
When adding new tests, it is a good habit to confirm a test would fail when the implementation is erroneous accordingly. But of course we cannot just rely on good habits. It is always better to be automated.

We actually had similar problems before #10749, or maybe some existing tests always pass (i.e. do not panic) even when the implementation violates what the test intends to cover, because the test itself is buggy.

Just sharing some thought. Not supporting or opposing the approach here.

Reply:
I now have a stupid tool 🤡 https://github.com/xxchan/cargo-utests/blob/main/src/main.rs

It's precise but slow (need to compile to know whether there are tests...) so we can't put it in CI. Alternatively, we can grep `#[test]` and mark `test=false` more conservatively. In this case, we can't disable binary targets which should be slow to compile.

Reply:
Oh, if the project's layout is very standard. (no `autobins=false`, all binaries are simple single-file wrappers), I think we can do better.

Reply:
Updated with only a few crates. They are heavy and probably won't have tests. Maybe we can start with this. 🤡 @BugenZhao